CloudinaryDotNet.IntegrationTests/UploadApi/UploadMethodsTest.cs

```diff
@@ -655,6 +656,129 @@ public async Task TestUploadLargeAutoFilesAsync()
 
             Assert.AreEqual("image", result.ResourceType);
         }
+
+        [Test, RetryWithDelay]
+        public void TestUploadChunkSingleStream()
+        {
+            var largeFilePath = m_testLargeImagePath;
+            var largeFileLength = (int)new FileInfo(largeFilePath).Length;
+
+            ImageUploadResult result = null;
+
+            using (var currChunk = new MemoryStream())
+            {
+                var uploadParams =  new ImageUploadParams()
+                {
+                    File = new FileDescription($"ImageFromChunks_{GetTaggedRandomValue()}", currChunk),
+                    Tags = m_apiTag
+                };
+
+                var buffer = new byte[TEST_CHUNK_SIZE];
+
+                using (var source = File.Open(largeFilePath, FileMode.Open))
+                {
+                    int read;
+                    while ((read = source.Read(buffer, 0, buffer.Length)) > 0)
+                    {
+                        currChunk.Write(buffer, 0, read);
+
+                        // Need to specify whether the chunk is the last one in order to finish the upload.
+                        uploadParams.File.LastChunk = read != TEST_CHUNK_SIZE;
```

Comment:
Theoretically, last chunk could be the same as all other chunks

Reply:
That's why this code is in unit test, in reality people will mark their last chunk in their way.